milli/src/search/matches/matching_words.rs

```diff
@@ -91,10 +105,13 @@ impl PartialEq for MatchingWord {
 }
 
 impl MatchingWord {
-    pub fn new(word: String, typo: u8, prefix: IsPrefix) -> Self {
+    pub fn new(word: String, typo: u8, prefix: IsPrefix) -> Option<Self> {
+        if word.len() > MAX_WORD_LENGTH {
+            return None;
+        }
```

Comment:
Can you think of another condition to add here that would avoid creating more useless matching word DFA?

Reply:
I'm afraid no, all I am thinking imply fetching data in the index